Solution1/ConsoleApp/Configuration/FileConfig.cs

```diff
@@ -5,12 +5,61 @@ namespace ConsoleApp.Configuration
 {
     public class FileConfig : IConfig
     {
-        public int MinCountDaysForecast => int.TryParse(ConfigurationManager.AppSettings["minCountDays"], out var value) ? value : default;
+        public AppConfiguration AppConfig { get => new AppConfiguration(); }
 
-        public int MaxCountDaysForecast => int.TryParse(ConfigurationManager.AppSettings["maxCountDays"], out var value) ? value : default;
+        public WeatherApiConfiguration ApiConfig { get => new WeatherApiConfiguration(); }
 
-        public bool IsDebugMode => bool.TryParse(ConfigurationManager.AppSettings["isDebugMode"], out var value) ? value : default;
+        public DbConfiguration DbConfig { get => new DbConfiguration(); }
+    }
+
+    public class AppConfiguration
+    {
+        public int MinCountDaysForecast { get; set; }
+
+        public int MaxCountDaysForecast { get; set; }
+
+        public bool IsDebugMode { get; set; }
+
+        public int? RequestTimeout { get; set; }
+
+        public AppConfiguration()
+        {
```

Comment:
you break single responsibility principle - configuration as dto | model is resposnsible for keeping info, you don't put any logic to constructures, why you do it here?